File: android/app/src/main/java/com/diego/pocketlink/protocol/MessageType.kt
package com.diego.pocketlink.protocol

enum class MessageType(val id: UShort) {
    HANDSHAKE(0x0001u),
    PING(0x0002u),
    PONG(0x0003u),
    DEVICE_INFO(0x0004u),
    ERROR(0x0005u),
    CLIPBOARD(0x0010u),
    BATTERY(0x0020u),
    NOTIFICATION(0x0030u),
    NOTIFICATION_REPLY(0x0031u),
    FILE_HEADER(0x0040u),
    FILE_CHUNK(0x0041u),
    FILE_ACK(0x0042u),
    FILE_CANCEL(0x0043u);

    companion object {
        fun fromId(id: UShort): MessageType? = entries.find { it.id == id }
    }
}

File: macos/PocketLinkCore/Sources/LinkProtocol/MessageType.swift
public enum MessageType: UInt16, Sendable, CaseIterable {
    case handshake = 0x0001
    case ping = 0x0002
    case pong = 0x0003
    case deviceInfo = 0x0004
    case error = 0x0005
    case clipboard = 0x0010
    case battery = 0x0020
    case notification = 0x0030
    case notificationReply = 0x0031
    case fileHeader = 0x0040
    case fileChunk = 0x0041
    case fileAck = 0x0042
    case fileCancel = 0x0043

    public init?(id: UInt16) {
        self.init(rawValue: id)
    }
}
